pkg/etcd/etcd_test.go

```diff
@@ -0,0 +1,302 @@
+		setup    func(*config.Control) error
+		teardown func(*config.Control) error
+		want     bool
+		wantErr  bool
+	}{
+		{
+			name: "Directory exists",
+			args: args{
+				ctx:    context.TODO(),
+				config: generateTestConfig(),
+			},
+			setup: func(cnf *config.Control) error {
+				if err := unit_tests.GenerateTestDataDir(cnf); err != nil {
+					return err
+				}
+				err := os.MkdirAll(walDir(cnf), 0700)
+				return err
+			},
+			teardown: func(cnf *config.Control) error {
+				unit_tests.CleanupTestDataDir(cnf)
+				err := os.Remove(walDir(cnf))
+				return err
+			},
+			wantErr: false,
+			want:    true,
+		},
+		{
+			name: "Directory does not exist",
+			args: args{
+				ctx:    context.TODO(),
+				config: generateTestConfig(),
+			},
+			setup: func(cnf *config.Control) error {
+				if err := unit_tests.GenerateTestDataDir(cnf); err != nil {
+					return err
+				}
+				// We don't care if removal fails to find the dir
+				os.Remove(walDir(cnf))
+				return nil
+			},
+			teardown: func(cnf *config.Control) error {
+				unit_tests.CleanupTestDataDir(cnf)
+				return nil
+			},
+			wantErr: false,
+			want:    false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			e := NewETCD()
+			defer tt.teardown(tt.args.config)
+			if err := tt.setup(tt.args.config); err != nil {
+				t.Errorf("Prep for ETCD.IsInitialized() failed = %v", err)
+				return
+			}
+			got, err := e.IsInitialized(tt.args.ctx, tt.args.config)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("ETCD.IsInitialized() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got != tt.want {
+				t.Errorf("ETCD.IsInitialized() = %+v\nWant = %+v", got, tt.want)
+				return
+			}
+		})
+	}
+}
+
+func TestETCD_Register(t *testing.T) {
+	type args struct {
+		ctx     context.Context
+		config  *config.Control
+		handler http.Handler
+	}
+	tests := []struct {
+		name     string
+		args     args
+		setup    func(cnf *config.Control) error
+		teardown func(cnf *config.Control) error
+		wantErr  bool
+	}{
+		{
+			name: "Call Register with standard config",
+			args: args{
+				ctx:     context.TODO(),
+				config:  generateTestConfig(),
+				handler: generateTestHandler(),
+			},
+			setup: func(cnf *config.Control) error {
+				err := unit_tests.GenerateTestRuntime(cnf)
+				return err
+			},
+			teardown: func(cnf *config.Control) error {
+				unit_tests.CleanupTestDataDir(cnf)
+				return nil
+			},
+		},
+		{
+			name: "Call Register with a tombstone file created",
+			args: args{
+				ctx:     context.TODO(),
+				config:  generateTestConfig(),
+				handler: generateTestHandler(),
+			},
+			setup: func(cnf *config.Control) error {
+				if err := unit_tests.GenerateTestRuntime(cnf); err != nil {
+					return err
+				}
+				if err := os.MkdirAll(etcdDBDir(cnf), 0700); err != nil {
+					return err
+				}
+				tombstoneFile := filepath.Join(etcdDBDir(cnf), "tombstone")
+				if _, err := os.Create(tombstoneFile); err != nil {
```

Comment:
```suggestion
				return os.Create(tombstoneFile); err != nil {
```

Reply:
In this and other areas with the same format I have changed to the suggested "return FUNC" format

Reply:
I made this comment in error. Disregard.